src/ec/packages/ec-component-button/ecl-button.html.twig

```diff
@@ -61,14 +61,20 @@
       {% if _icon_position == 'before' %}
         {% include '@ecl-twig/ec-component-icon/ecl-icon.html.twig' with {
           icon: _icon,
-          extra_classes: 'ecl-button__icon ecl-button__icon--before'
+          extra_classes: 'ecl-button__icon ecl-button__icon--before',
+          extra_attributes: [
+            { name: 'data-ecl-icon', value: 'true' }
```

Comment:
So....our approach with boolean attributes follows the html 5 defintion:
boolean attributes should be defined by their name only, setting a "true" value doesn't make them boolean, true is a string and unfortunately we see these custom booleans in ECL always with a quoted value which is basically wrong, so we don't have to reproduce that.
The only use case where we would set the "true" value as a string is when it is used in javascript (which should be avoided).
The alternative would be to set the true value unquoted in order for it to be a boolean, but it's confusing, that is also why we try to handle these differences with ECL when we make a diff.